### PR TITLE
Update ExaMiniMD for Kokkos v3.0 promotion

### PR DESCRIPTION
**Pull request by alanphumphrey**
This pull request updates ExaMiniMD for the Kokkos version 3.0 promotion, and works with deprecated code disabled (KOKKOS_OPTIONS= disable_deprecated_code). There will be a complete removal of deprecated code support after the Kokkos version 3.0 release, hence this pull request. Current status and timeline for Kokkos code deprecations is at: https://github.com/kokkos/kokkos/wiki/DeprecationPage

This pull request corresponds to issue #27.

Please note that with this pull request, ExaMiniMD no longer builds with deprecated code enabled (KOKKOS_OPTIONS=enable_deprecated_code), due to the implementation of StaticCrsGraph (Kokkos_StaticCrsGraph.hpp) not using variadic templates (e.g., order of template arguments matters).

### CODE REVIEW

File: src/force_types/force_snap_neigh_impl.h [hunk 1]
```diff
@@ -172,15 +172,15 @@ void ForceSNAP<NeighborClass>::compute(System* system, Binning* binning, Neighbo
...
+  Kokkos::parallel_reduce("ForceSNAP::find_max_neighs",nlocal, FindMaxNumNeighs<t_neigh_list>(neigh_list), Kokkos::Max<int>(max_neighs));
 
   sna.nmax = max_neighs;
 
   T_INT team_scratch_size = sna.size_team_scratch_arrays();
   T_INT thread_scratch_size = sna.size_thread_scratch_arrays();
 
   //printf("Sizes: %i %i\n",team_scratch_size/1024,thread_scratch_size/1024);
-  int team_size_max = Kokkos::TeamPolicy<>::team_size_max(*this);
+  int team_size_max = Kokkos::TeamPolicy<>(1,1).team_size_max(*this,Kokkos::ParallelForTag());
```
Comment (stanmoore1) [inline src/force_types/force_snap_neigh_impl.h]: This seems really ad-hoc. Can this be made more precise?

Comment (alanphumphrey) [reply src/force_types/force_snap_neigh_impl.h]: Agreed. I had mentioned this in that specific commit. Will make this more precise.

Comment (stanmoore1) [reply src/force_types/force_snap_neigh_impl.h]: OK thanks @alanphumphrey. The other changes look good.

Comment (alanphumphrey) [reply src/force_types/force_snap_neigh_impl.h]: @stanmoore1 - For clarity, as of Kokkos v3.0, the team_size_max is now a member function of the team policy. The previous variant didn't take all necessary information into account and could result in invalid answers.

In obtaining the team_size_max value, a Kokkos::TeamPolicy<> instance is now created in your code (force_types/force_snap_neigh_impl.h:183), using the default execution space and launching a league with "league_size" teams with "team_size" threads each.

As the team_size_max call is specifically to determine that value, we use Kokkos::AUTO to pick the "team_size" arg. As long as the loop is simpler with only few variables, Kokkos::AUTO should work well. We can then use nlocal for the "league_size" arg (as you've done in force_types/force_snap_neigh_impl.h:192).

Although these changes are more precise and should work well, let me know if this still seems ad-hoc. You may consider slightly reorganizing void ForceSNAP<NeighborClass>::compute based on this input (without the anonymous policy instance). I'm happy to help.

For example, with Nvidia GPUs you might want to consider trying the number SMs for league_size. For say, Kokkos::OpenMP, a reasonable starting point might be a league size of <#_of_cores> and a team size between <1_to_#_of_threads_per_core>. Our group (Uintah AMT runtime system) has experimented with this quite a bit.

This optimal performance balancing act is mostly just about having enough threads to fill to either be compute, memory, or latency bounded.

**Comment by stanmoore1** [comment]
@alanphumphrey will this build with Kokkos 2.9.0, or do I need to wait for Kokkos 3.0? Thanks for the help.

**Comment by alanphumphrey** [comment]
@stanmoore1, yes this will compile with Kokkos v2.9, however you need to explicitly disable support for deprecated code, e.g. set KOKKOS_OPTIONS=disable_deprecated_code. In ExaMiniMD/src/makefile, change line 35 from:

KOKKOS_OPTIONS=aggressive_vectorization
      to
KOKKOS_OPTIONS=aggressive_vectorization,disable_deprecated_code

This is because deprecated code support is ON by default in v2.9. With 3.0 all this support will be removed and my changes will work by default.

Happy to help, let me know if you have any other questions.

**Pull request merged**